### ISSUE TITLE
[ban-types] False detection of {} on default arg values

### ISSUE THREAD

**Issue by MichalLytek** [typescript-eslint/typescript-eslint#2078]
**Repro**

I follwed the updgrade guide to v3.0 and introduced the type that can replace `object`:

```ts
export type AnyObject = Record<string | symbol, unknown>;
```
And then I replaced all of the places with `object` usage like `<TContext extends object = {}>`:

```ts
export interface BuildSchemaConfig<TContext extends AnyObject = {}>
```

But it now reports the error on runtime default value (new empty object):

![image](https://user-images.githubusercontent.com/10618781/82731355-79613600-9d06-11ea-9653-dc7cb8d7547f.png)

```
Don't use `{}` as a type. `{}` actually means "any non-nullish value".
- If you want a type meaning "any object", you probably want `Record<string, unknown>` instead.
- If you want a type meaning "any value", you probably want `unknown` instead.eslint(@typescript-eslint/ban-types)
```

I am using the default settings from `plugin:@typescript-eslint/recommended` for this.

**Expected Result**

No error

**Actual Result**

Error about runtime default value of empty object

**Additional Info**

**Versions**

| package                            | version   |
| ---------------------------------- | --------- |
| `@typescript-eslint/eslint-plugin` | `3.0.0`   |
| `@typescript-eslint/parser`        | `3.0.0`   |
| `TypeScript`                       | `3.9.3`   |
| `ESLint`                           | `7.1.0`   |
| `node`                             | `13.12.0` |
| `npm`                              | `6.14.4`  |

**Comment by MichalLytek** [comment]
Oh, it's the default type, not arg value 😄
